src/pages/ProfilePage.js

```diff
@@ -0,0 +1,240 @@
+        accountID: PropTypes.string,
+
+        /** report ID passed */
+        reportID: PropTypes.string,
+    }),
+});
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** The personal details of all users */
+    personalDetails: PropTypes.objectOf(personalDetailsPropType),
+
+    /** Route params */
+    route: matchType.isRequired,
+
+    /** Login list for the user that is signed in */
+    loginList: PropTypes.shape({
+        /** Phone/Email associated with user */
+        partnerUserID: PropTypes.string,
+    }),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    // When opening someone else's profile (via deep link) before login, this is empty
+    personalDetails: {},
+    loginList: {},
+};
+
+/**
+ * Gets the phone number to display for SMS logins
+ *
+ * @param {Object} details
+ * @param {String} details.login
+ * @param {String} details.displayName
+ * @returns {String}
+ */
+const getPhoneNumber = (details) => {
+    // If the user hasn't set a displayName, it is set to their phone number, so use that
+    const displayName = lodashGet(details, 'displayName', '');
+    const parsedPhoneNumber = parsePhoneNumber(displayName);
+    if (parsedPhoneNumber.possible) {
+        return parsedPhoneNumber.number.e164;
+    }
+
+    // If the user has set a displayName, get the phone number from the SMS login
+    return details.login ? Str.removeSMSDomain(details.login) : '';
+};
+
+function ProfilePage(props) {
+    const accountID = lodashGet(props.route.params, 'accountID', 0);
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    useEffect(() => {
+        if (accountID > 0) {
+            PersonalDetails.openPublicProfilePage(accountID);
+        }
+    }, [accountID]);
+
+    const details = lodashGet(props.personalDetails, accountID, {});
+    const displayName = details.displayName ? details.displayName : props.translate('common.hidden');
+    const avatar = lodashGet(details, 'avatar', UserUtils.getDefaultAvatar());
+    const originalFileName = lodashGet(details, 'originalFileName', '');
+    const login = lodashGet(details, 'login', '');
+    const timezone = lodashGet(details, 'timezone', {});
+
+    // If we have a reportID param this means that we
+    // arrived here via the ParticipantsPage and should be allowed to navigate back to it
+    const shouldShowLocalTime = !ReportUtils.hasAutomatedExpensifyEmails([login]) && !_.isEmpty(timezone);
+
+    let pronouns = lodashGet(details, 'pronouns', '');
+    if (pronouns && pronouns.startsWith(CONST.PRONOUNS.PREFIX)) {
+        const localeKey = pronouns.replace(CONST.PRONOUNS.PREFIX, '');
+        pronouns = props.translate(`pronouns.${localeKey}`);
+    }
+
+    const isSMSLogin = Str.isSMSLogin(login);
+    const phoneNumber = getPhoneNumber(details);
+    const phoneOrEmail = isSMSLogin ? getPhoneNumber(details) : login;
+
+    const isCurrentUser = _.keys(props.loginList).includes(login);
```

Comment:
We should have used the account ID to check if is current user because:
- Users are now identified by account ids
- The account ID is available from the route where the login is from onyx. Using data from onyx results in a delay where the first render will always evaluated `isCurrentUser` as false and causing this https://github.com/Expensify/App/issues/28591

Reply:
Ah good catch, thanks for letting me know!